tests/functional/test_install.py

```diff
@@ -1527,3 +1527,46 @@ def test_target_install_ignores_distutils_config_install_prefix(script):
         (target - script.base_path) in result.files_created and
         (prefix - script.base_path) not in result.files_created
     ), str(result)
+
+
+@pytest.mark.network
```

Comment:
It would be best to avoid adding more tests that depend on network access, IMO, as it adds flakiness, makes the tests slower, etc. It would be best if we can find the appropriate places at which to unit test to test this functionality (starting with the "inner-most" functions).